src/test/codewhisperer/util/crossFileContextUtil.test.ts

```diff
@@ -0,0 +1,109 @@
+            assert.strictEqual(actual, 6)
+        })
+
+        it('distance 7', function () {
+            candidateFile = 'ui/popup/components/CodeWhispererPopup.java'
+            const actual = getFileDistance(targetFile, candidateFile, '/')
+            assert.strictEqual(actual, 7)
+        })
+
+        it('distance 8', function () {
+            candidateFile = 'ui/popup/components/actions/AcceptRecommendationAction.java'
+            const actual = getFileDistance(targetFile, candidateFile, '/')
+            assert.strictEqual(actual, 8)
+        })
+    })
+
+    describe('getRelevantFiles', function () {
+        let tempFolder: string
+
+        beforeEach(async function () {
+            tempFolder = await makeTemporaryToolkitFolder()
+        })
+
+        afterEach(async function () {
+            await fs.remove(tempFolder)
+        })
+
+        it('should return opened files in the current window and sorted ascendingly by file distance', async function () {
+            const targetFile = path.join(tempFolder, 'service', 'microService', 'CodeWhispererFileContextProvider.java')
+            const fileWithDistance3 = path.join(tempFolder, 'service', 'CodewhispererRecommendationService.java')
+            const fileWithDistance5 = path.join(tempFolder, 'util', 'CodeWhispererConstants.java')
+            const fileWithDistance6 = path.join(tempFolder, 'ui', 'popup', 'CodeWhispererPopupManager.java')
+            const fileWithDistance7 = path.join(tempFolder, 'ui', 'popup', 'components', 'CodeWhispererPopup.java')
+            const fileWithDistance8 = path.join(
+                tempFolder,
+                'ui',
+                'popup',
+                'components',
+                'actions',
+                'AcceptRecommendationAction.java'
+            )
+
+            const filePaths = [
+                targetFile,
+                fileWithDistance8,
+                fileWithDistance5,
+                fileWithDistance7,
+                fileWithDistance3,
+                fileWithDistance6,
+            ]
+            const shuffledFilePaths = shuffleList(filePaths)
+
+            for (const file of shuffledFilePaths) {
+                toFile('', file)
+                const textDocument = await vscode.workspace.openTextDocument(vscode.Uri.file(file))
+                await vscode.window.showTextDocument(textDocument, { preview: false })
+            }
+
+            // to make the target file editor active
+            const textDocument = await vscode.workspace.openTextDocument(vscode.Uri.file(targetFile))
+            const editor = await vscode.window.showTextDocument(textDocument)
+
+            const actual = await getRelevantCrossFiles(editor)
+            assert.deepStrictEqual(actual, [
+                fileWithDistance3,
+                fileWithDistance5,
+                fileWithDistance6,
+                fileWithDistance7,
+                fileWithDistance8,
+            ])
+        })
+    })
+})
```

Comment:
another q for @JadenSimon @justinmk3 is this the correct way to 
1. set up a project structure in test 
2. open them in the editors VSC

since the CI succeeded on some machines and failed in others 
https://d1ihu6zq92vp9p.cloudfront.net/c2cc476c-5da9-4d4b-b5cd-86a6ac88f0a9/report.html

Reply:
thanks!

Reply:
`vscode.workspace.openTextDocument` / `showTextDocument` may be racy, you need to wait for the IDE: 

https://github.com/aws/aws-toolkit-vscode/blob/d66f115ee4e1fed8d87fa6a59bd83a1c5925ac05/src/test/testUtil.ts#L221

Reply:
gotcha thanks Justin!

Reply:
for my own education, i thought `await` will wait for the async to complete and continue?

Reply:
why in this case it's racy

Reply:
@justinmk3 do you know why there is still race condition even I add the assert clause there lol... I have tried all the combinations but the linx & mac tests still failed ...

Reply:
`await showTextDocument` doesn't make any guarantees about the IDE state (text editors). That's why `assertTextEditorContains` uses `waitUntil` to retry a condition for a maximum time.

Reply:
do you think this will work @justinmk3 
```
export async function assertTabSize(size: number): Promise<void | never> {
    const tabs = await waitUntil(
        async () => {
            const tabs = vscode.window.tabGroups.all
                .map(tabGroup => tabGroup.tabs)
                .reduce((acc, curVal) => acc.concat(curVal), [])

            if (tabs.length === size) {
                return tabs
            }
        },
        { interval: 5 }
    )

    if (!tabs) {
        throw new Error('No desired tabs found')
    }
}
```